clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -5407,6 +5407,9 @@ def note_dependent_function_template_spec_discard_reason : Note<
   "candidate ignored: %select{not a function template|"
   "not a member of the enclosing %select{class template|"
   "namespace; did you mean to explicitly qualify the specialization?}1}0">;
+def warn_diag_specialization : Warning<
+  "specializing a template which should not be specialized">,
```

Comment:
I fee; like rephrasing it to something along the lines of ‘'X' should not be specialised’ would make it a bit clearer what the problem is.

‘Specialising a template which should not be specialised’ is probably a bit unclear to anyone who doesn’t already know about this rule, whereas e.g. ‘`std::something` should not be specialised’ imo makes it clearer what exactly the problem here is (i.e. that *particular* template should not be specialised).